SU-HSDO/suhumsci
# STN-490:  Refactor Grid System

#  READY FOR REVIEW

## Summary
Update grid system to use CSS Grid instead of Flexbox. This update removes the extra margin bottom spacing found on grid items. This is especially noticeable in row wells.

Adjusts grid views AND `.counted-item` rows where flexbox is being used.

Reference: JIRA [STN-490](https://sparkbox.atlassian.net/browse/STN-490)
Meets the following acceptance criteria:
 - [ ] CSS Grid system has been tested to work in IE11.
 - [ ] Grid system works in all modern browsers
 - [ ] Grid items in row wells no longer have extra spacing between the last row of grid items and the parent container.
 - [ ] We are not introducing new regressions (run backstop.js)

## Need Review By (Date)
8/19/2020

## Urgency
medium

## Steps to Test to grid views
1. Review the grid layout on the items on the following pages and compare them against the Sparkbox Sandbox [staging site](https://sparkbox-sandbox-dev.stanford.edu/). At mobile screens grid items should stack into a single column. At the small breakpoint, grid items should stack into 2 columns. At the medium breakpoint, grid items using 3 or more columns should stack into 3 columns. And at the large breakpoint, grid items using 4 or more columns should stack into the correct number of columns as declared by `.hb-grid--cols-#`.
 - [x] http://sparkbox-sandbox.suhumsci.loc/
    * Cards (grid items) in the `Spotlights` section.
    * Cards inside of the well beneath the `Spotlights` section.
    * Events cards beneath `This week's events, seminars and graduate student workshops`
- [x] http://sparkbox-sandbox.suhumsci.loc/people/people-grid
- [x] http://sparkbox-sandbox.suhumsci.loc/publications/book-shelf
 - [x] http://sparkbox-sandbox.suhumsci.loc/people/qa-person-all-fields-filled
    * Cards under `Latest Publications`
2. The max number of columns that can be set for the grid is 12. Although we don't recommend using a 12 column grid, we should still test and make sure all grid column possibilities work. To do this I recommend using the home page. http://sparkbox-sandbox.suhumsci.loc/
    * Change the number of cards displayed under `Spotlights` from 4 to 12. Under `Spotlights` edit the view. Find `Pager`. Change it to 12 items and save the view. 
![4 to 12](https://user-images.githubusercontent.com/12678977/90526430-63fd7b80-e13e-11ea-8845-f2ee99c46092.png)
    * Go back to the home page. Open your browser dev tools. Find the `hb-grid hb-grid--cols-4` classes on the containing div for the cards under `Spotlights`. Change `hb-grid--cols-4` to `hb-grid--cols-5`. You should now see 5 columns. Expand / reduce the browser window to test breakpoints to make sure 5 columns at full screen drops to 3 columns, then 2 columns, and then 1 at mobile. Repeat for 6, 7, 8, 9, 10, 11, and 12 columns.

## Steps to Test to `counted-item` rows
1. Review the grid layout on the items on the following pages and compare them against the Sparkbox Sandbox [staging site](https://sparkbox-sandbox-dev.stanford.edu/).
- [x] http://sparkbox-sandbox.suhumsci.loc/
- [x] http://sparkbox-sandbox.suhumsci.loc/texts-cards-row-wells
    * test vertical cards
    * test horizontal cards
    * 2 column combinations of event cards and accordions
    * vertical linked cards
- [x] http://sparkbox-sandbox.suhumsci.loc/qa/qa-postcards
    * vertical cards
    * vertical links cards
- [x] http://sparkbox-sandbox.suhumsci.loc/qa/qa-stretch-vertical-linked-postcards
    * vertical linked cards
 - [x] http://sparkbox-sandbox.suhumsci.loc/qa/qa-page
    * 2 column combination of image w/ WYSIWYG text
    * 4 vertical cards in a row
    * 2 columns of text area right next to each other in a row
